features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverSnmp4jImpl.java

```diff
@@ -150,19 +168,93 @@ public void stop() throws InterruptedException {
     /**
      * The execution context.
      */
-    // TODO: HZN-609: Change this to TrapdNotificationListener.receive()
     @Override
     public void run() {
-        SyslogConnection connection = new SyslogConnection(pkt, m_config);
+    }
 
+	@Override
+	public void trapReceived(TrapNotification trapNotification) {
+		try {
+			  for (TrapNotificationHandler handler : m_trapNotificationHandlers) {
+			    handler.handleTrapNotification(trapNotification);
+			  }
+			} catch (Throwable e) {
+			  LOG.error("Handler execution failed in {}", this.getClass().getSimpleName(), e);
+			}
+	}
+
+	@Override
+	public void trapError(int error, String msg) {
+      LOG.warn("Error Processing Received Trap: error = {} {}", error, (msg != null ? ", ref = " + msg : ""));
+	}
+	
+	public void registeredForTraps(){
         try {
-            for (TrapNotificationHandler handler : m_trapNotificationHandlers) {
-                handler.handleTrapNotification(connection);
+        	InetAddress address = getInetAddress();
+        	LOG.info("Listening on {}:{}", address == null ? "[all interfaces]" : InetAddressUtils.str(address), m_snmpTrapPort);
+            SnmpUtils.registerForTraps(this, this, address, m_snmpTrapPort, m_snmpV3Users); // Need to clarify 
+            m_registeredForTraps = true;
+            
+            LOG.debug("init: Creating the trap session");
+        } catch (final IOException e) {
+            if (e instanceof java.net.BindException) {
+                Logging.withPrefix("OpenNMS.Manager", new Runnable() {
+                    @Override
+                    public void run() {
+                        LOG.error("init: Failed to listen on SNMP trap port, perhaps something else is already listening?", e);
+                    }
+                });
+                LOG.error("init: Failed to listen on SNMP trap port, perhaps something else is already listening?", e);
+            } else {
+                LOG.error("init: Failed to initialize SNMP trap socket", e);
             }
-        } catch (Throwable e) {
-            LOG.error("Handler execution failed in {}", this.getClass().getSimpleName(), e);
+            throw new UndeclaredThrowableException(e);
         }
-    }
+	}
+	
+	public void unRegisteredForTraps(){
```

Comment:
Change this method name to stop() and add an @Override annotation to indicate that it overrides an interface method inside TrapReceiver.